ModelSkin.cpp, ambiguous operator compilation issue

This is a compilation issue:
On Visual Studio, but also seen on other compilers apparently, the following error message occurs.
```Severity	Code	Description	Project	File	Line	Suppression State	Detail Description
Error (active)	E0350	more than one operator "=" matches these operands:	scenegraph	D:\Code\PioneerGit\pioneer_fluffy\src\scenegraph\ModelSkin.cpp	114		            function "std::basic_string<_Elem, _Traits, _Alloc>::operator=(std::basic_string<_Elem, _Traits, _Alloc> &&_Right) [with _Elem=char, _Traits=std::char_traits<char>, _Alloc=std::allocator<char>]"
            function "std::basic_string<_Elem, _Traits, _Alloc>::operator=(std::initializer_list<_Elem> _Ilist) [with _Elem=char, _Traits=std::char_traits<char>, _Alloc=std::allocator<char>]"
            operand types are: std::string = nlohmann::basic_json<std::map, std::vector, std::string, bool, int64_t, uint64_t, double, std::allocator, nlohmann::adl_serializer>
```

@Web-eWorks as usual Visual Studio is being super helpful there with it's template expansion.
It says that it's an error, and has been there since 3a3a13edec758fbfa9b1eae0e488a528ebb27468 however [the code](https://github.com/pioneerspacesim/pioneer/blob/master/src/scenegraph/ModelSkin.cpp#L114) still builds (_obviously_) and runs fine as far as I know.

Here is the complete error log:

```
1>cl : command line warning D9035: option 'Gm' has been deprecated and will be removed in a future release
1>ModelSkin.cpp
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\ucrt\corecrt_math_defines.h(22,1): warning C4005: 'M_PI': macro redefinition
1>D:\pioneer\build\pioneer-thirdparty\win32\include\SDL_stdinc.h(499): message : see previous definition of 'M_PI'
1>D:\pioneer\build\pioneer-ecraven-systemview-squashed\src\scenegraph\Serializer.h(48,1): warning C4267: 'argument': conversion from 'size_t' to 'const T', possible loss of data
1>        with
1>        [
1>            T=Uint32
1>        ]
1>D:\pioneer\build\pioneer-ecraven-systemview-squashed\src\scenegraph\Serializer.h(59,1): warning C4267: 'initializing': conversion from 'size_t' to 'Uint32', possible loss of data
1>D:\pioneer\build\pioneer-ecraven-systemview-squashed\src\scenegraph\Serializer.h(83,1): warning C4267: 'argument': conversion from 'size_t' to 'Uint32', possible loss of data
1>D:\pioneer\build\pioneer-ecraven-systemview-squashed\src\scenegraph\ModelSkin.cpp(114,37): error C2593: 'operator =' is ambiguous
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\xstring(2591,19): message : could be 'std::basic_string<char,std::char_traits<char>,std::allocator<char>> &std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator =(std::initializer_list<_Elem>)'
1>        with
1>        [
1>            _Elem=char
1>        ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\xstring(2492,19): message : or       'std::basic_string<char,std::char_traits<char>,std::allocator<char>> &std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator =(std::basic_string<char,std::char_traits<char>,std::allocator<char>> &&) noexcept'
1>D:\pioneer\build\pioneer-ecraven-systemview-squashed\src\scenegraph\ModelSkin.cpp(114,37): message : while trying to match the argument list '(std::string, nlohmann::basic_json<std::map,std::vector,std::string,bool,int64_t,uint64_t,double,std::allocator,nlohmann::adl_serializer>)'
1>D:\pioneer\build\pioneer-ecraven-systemview-squashed\src\scenegraph\ModelSkin.cpp(117,35): error C2593: 'operator =' is ambiguous
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\xstring(2591,19): message : could be 'std::basic_string<char,std::char_traits<char>,std::allocator<char>> &std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator =(std::initializer_list<_Elem>)'
1>        with
1>        [
1>            _Elem=char
1>        ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\xstring(2492,19): message : or       'std::basic_string<char,std::char_traits<char>,std::allocator<char>> &std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator =(std::basic_string<char,std::char_traits<char>,std::allocator<char>> &&) noexcept'
1>D:\pioneer\build\pioneer-ecraven-systemview-squashed\src\scenegraph\ModelSkin.cpp(117,35): message : while trying to match the argument list '(std::string, nlohmann::basic_json<std::map,std::vector,std::string,bool,int64_t,uint64_t,double,std::allocator,nlohmann::adl_serializer>)'
1>Done building project "newmodel.vcxproj" -- FAILED.
2>------ Build started: Project: pioneer, Configuration: Release x64 ------
2>cl : command line warning D9025: overriding '/sdl-' with '/GS-'
2>cl : command line warning D9025: overriding '/sdl-' with '/GS-'
2>cl : command line warning D9025: overriding '/sdl-' with '/GS-'
2>Body.cpp
2>Frame.cpp
2>cl : command line warning D9025: overriding '/sdl-' with '/GS-'
2>Ship.cpp
2>D:\pioneer\build\pioneer-ecraven-systemview-squashed\src\Frame.cpp(31,28): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
2>D:\pioneer\build\pioneer-ecraven-systemview-squashed\src\Frame.cpp(36,49): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data
2>D:\pioneer\build\pioneer-ecraven-systemview-squashed\src\Frame.cpp(61,28): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
2>D:\pioneer\build\pioneer-ecraven-systemview-squashed\src\Frame.cpp(170,30): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
2>D:\pioneer\build\pioneer-ecraven-systemview-squashed\src\Frame.cpp(195,33): error C2593: 'operator =' is ambiguous
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\xstring(2591,19): message : could be 'std::basic_string<char,std::char_traits<char>,std::allocator<char>> &std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator =(std::initializer_list<_Elem>)'
2>        with
2>        [
2>            _Elem=char
2>        ] (compiling source file ..\..\src\Frame.cpp)
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\xstring(2492,19): message : or       'std::basic_string<char,std::char_traits<char>,std::allocator<char>> &std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator =(std::basic_string<char,std::char_traits<char>,std::allocator<char>> &&) noexcept' (compiling source file ..\..\src\Frame.cpp)
2>D:\pioneer\build\pioneer-ecraven-systemview-squashed\src\Frame.cpp(195,33): message : while trying to match the argument list '(std::string, const nlohmann::basic_json<std::map,std::vector,std::string,bool,int64_t,uint64_t,double,std::allocator,nlohmann::adl_serializer>)'
2>D:\pioneer\build\pioneer-ecraven-systemview-squashed\src\Frame.cpp(261,30): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
2>D:\pioneer\build\pioneer-ecraven-systemview-squashed\src\Body.cpp(47,29): error C2593: 'operator =' is ambiguous
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\xstring(2591,19): message : could be 'std::basic_string<char,std::char_traits<char>,std::allocator<char>> &std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator =(std::initializer_list<_Elem>)'
2>        with
2>        [
2>            _Elem=char
2>        ] (compiling source file ..\..\src\Body.cpp)
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\xstring(2492,19): message : or       'std::basic_string<char,std::char_traits<char>,std::allocator<char>> &std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator =(std::basic_string<char,std::char_traits<char>,std::allocator<char>> &&) noexcept(<expr>)' (compiling source file ..\..\src\Body.cpp)
2>D:\pioneer\build\pioneer-ecraven-systemview-squashed\src\Body.cpp(47,29): message : while trying to match the argument list '(std::string, nlohmann::basic_json<std::map,std::vector,std::string,bool,int64_t,uint64_t,double,std::allocator,nlohmann::adl_serializer>)'
2>D:\pioneer\build\pioneer-ecraven-systemview-squashed\src\Ship.cpp(191,31): error C2593: 'operator =' is ambiguous
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\xstring(2591,19): message : could be 'std::basic_string<char,std::char_traits<char>,std::allocator<char>> &std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator =(std::initializer_list<_Elem>)'
2>        with
2>        [
2>            _Elem=char
2>        ] (compiling source file ..\..\src\Ship.cpp)
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\xstring(2492,19): message : or       'std::basic_string<char,std::char_traits<char>,std::allocator<char>> &std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator =(std::basic_string<char,std::char_traits<char>,std::allocator<char>> &&) noexcept' (compiling source file ..\..\src\Ship.cpp)
2>D:\pioneer\build\pioneer-ecraven-systemview-squashed\src\Ship.cpp(191,31): message : while trying to match the argument list '(std::string, nlohmann::basic_json<std::map,std::vector,std::string,bool,int64_t,uint64_t,double,std::allocator,nlohmann::adl_serializer>)'
2>Done building project "pioneer.vcxproj" -- FAILED.
========== Build: 0 succeeded, 2 failed, 13 up-to-date, 0 skipped ==========
```
The compiler swears at:
https://github.com/pioneerspacesim/pioneer/blob/cde7642711dc808a2d06a384584da2466e555af2/src/Body.cpp#L47
https://github.com/pioneerspacesim/pioneer/blob/cde7642711dc808a2d06a384584da2466e555af2/src/Frame.cpp#L195
https://github.com/pioneerspacesim/pioneer/blob/cde7642711dc808a2d06a384584da2466e555af2/src/scenegraph/ModelSkin.cpp#L114
https://github.com/pioneerspacesim/pioneer/blob/cde7642711dc808a2d06a384584da2466e555af2/src/scenegraph/ModelSkin.cpp#L117
https://github.com/pioneerspacesim/pioneer/blob/cde7642711dc808a2d06a384584da2466e555af2/src/Ship.cpp#L191

I would like to clarify that I personally do not have such a problem.
(gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0)

--- a/src/Body.cpp
+++ b/src/Body.cpp
@@ -44,7 +44,7 @@ Body::Body(const Json &jsonObj, Space *space) :
 
 		Properties().LoadFromJson(bodyObj);
 		m_frame = bodyObj["index_for_frame"];
-		m_label = bodyObj["label"];
+		m_label = bodyObj["label"].get<std::string>();
 		Properties().Set("label", m_label);
 		m_dead = bodyObj["dead"];
 

--- a/src/Frame.cpp
+++ b/src/Frame.cpp
@@ -192,7 +192,7 @@ FrameId Frame::FromJson(const Json &frameObj, Space *space, FrameId parent, doub
 
 		f->m_flags = frameObj["flags"];
 		f->m_radius = frameObj["radius"];
-		f->m_label = frameObj["label"];
+		f->m_label = frameObj["label"].get<std::string>();
 
 		f->m_pos = frameObj["pos"];
 		f->m_angSpeed = frameObj["ang_speed"];

--- a/src/Ship.cpp
+++ b/src/Ship.cpp
@@ -188,7 +188,7 @@ Ship::Ship(const Json &jsonObj, Space *space) :
 
 		m_navLights->LoadFromJson(shipObj);
 
-		m_shipName = shipObj["name"];
+		m_shipName = shipObj["name"].get<std::string>();
 		Properties().Set("shipName", m_shipName);
 	} catch (Json::type_error &) {
 		throw SavedGameCorruptException();

--- a/src/scenegraph/ModelSkin.cpp
+++ b/src/scenegraph/ModelSkin.cpp
@@ -111,10 +111,10 @@ namespace SceneGraph {
 			if (decalsArray.size() != MAX_DECAL_MATERIALS) throw SavedGameCorruptException();
 			for (unsigned int i = 0; i < MAX_DECAL_MATERIALS; i++) {
 				Json arrayElem = decalsArray[i];
-				m_decals[i] = arrayElem["decal"];
+				m_decals[i] = arrayElem["decal"].get<std::string>();
 			}
 
-			m_label = modelSkinObj["label"];
+			m_label = modelSkinObj["label"].get<std::string>();
 		} catch (Json::type_error &) {
 			throw SavedGameCorruptException();
 		}

